src/main/java/io/xpring/xrpl/Utils.java

```diff
@@ -1,6 +1,8 @@
 package io.xpring.xrpl;
 
 import com.google.common.base.Preconditions;
+import io.xpring.codec.addresses.ClassicAddressCodec;
```

Comment:
This class was meant to act as a facade, which simply wrapped another class (`JavaScriptUtils`, `CppUtils`). In a very over-engineered world, you could imagine refactoring this class to use some sort of ServiceRegistry/Dependency Injection to pull a native, C++, or JS implementation based on the environment. 

That said, it was definitely a pipe dream to set up a system like described above, and admittedly it may have been over engineered (and the design you are proposing does not preclude that future either). Adding your implementation to a UtilsImpl.java class might preserve that future, but it is up to you whether that is interesting to you.

Reply:
Yeah, some sort of DI or service registry is going to be needed once there are multiple implementations and to make unit testing easier. For now, I wasn't going to tackle that since it doesn't feel like a big issue.